package.json

```diff
@@ -84,27 +85,203 @@
     }
   },
   "contributes": {
+    "icons": {
+      "camel-run": {
+        "description": "Run Camel Application",
+        "default": {
+          "fontPath": "./icons/camel-run.woff",
+          "fontCharacter": "\\E900"
+        }
+      }
+    },
     "commands": [
       {
         "command": "kaoto.open",
-        "title": "Open with Kaoto Graphical Editor for Camel"
+        "title": "Open with Kaoto"
       },
       {
-        "command": "kaoto.open.textualeditor",
-        "title": "Open Camel file with textual editor on the side",
-        "shortTitle": "Show source",
+        "command": "kaoto.open.source",
+        "title": "Open Source Code",
+        "shortTitle": "Show Source",
         "enablement": "activeCustomEditorId == webviewEditorsKaoto",
-        "category": "Camel",
-        "icon": "$(go-to-file)"
+        "category": "Kaoto",
+        "icon": "$(code)"
+      },
+      {
+        "command": "kaoto.close.source",
+        "title": "Close Source Code",
+        "shortTitle": "Close Source",
+        "enablement": "resourceFilename =~ /\\.(yaml)$/",
+        "category": "Kaoto",
+        "icon": "$(code)"
+      },
```

Comment:
hen called from command palette it is closing whatever editor is opened, even the kaoto one. If we keep this command, need to filter when to display it in the command palette